src/NHibernate/Cache/CacheFactory.cs

```diff
@@ -30,17 +30,41 @@ public static class CacheFactory
 		/// <param name="settings">Used to retrieve the global cache region prefix.</param>
 		/// <param name="properties">Properties the cache provider can use to configure the cache.</param>
 		/// <returns>An <see cref="ICacheConcurrencyStrategy"/> to use for this object in the <see cref="ICache"/>.</returns>
-		public static ICacheConcurrencyStrategy CreateCache(string usage, string name, bool mutable, Settings settings,
-		                                                    IDictionary<string,string> properties)
+		// Since v5.1
+		[Obsolete("Please use overload with an ICache parameter.")]
+		public static ICacheConcurrencyStrategy CreateCache(
+			string usage,
+			string name,
+			bool mutable,
+			Settings settings,
+			IDictionary<string, string> properties)
 		{
-			if (usage == null || !settings.IsSecondLevelCacheEnabled) return null; //no cache
+			return CreateCache(
+				usage, name, mutable, settings,
+				r => settings.CacheProvider.BuildCache(r, properties));
+		}
 
-			string prefix = settings.CacheRegionPrefix;
-			if (prefix != null) name = prefix + '.' + name;
+		/// <summary>
+		/// Creates an <see cref="ICacheConcurrencyStrategy"/> from the parameters.
+		/// </summary>
+		/// <param name="usage">The name of the strategy that <see cref="ICacheProvider"/> should use for the class.</param>
+		/// <param name="name">The name of the class the strategy is being created for.</param>
+		/// <param name="mutable"><see langword="true" /> if the object being stored in the cache is mutable.</param>
+		/// <param name="settings">Used to retrieve the global cache region prefix.</param>
+		/// <param name="regionCacheGetter">The delegate for obtaining the <see cref="ICache" /> to use for the region.</param>
+		/// <returns>An <see cref="ICacheConcurrencyStrategy"/> to use for this object in the <see cref="ICache"/>.</returns>
+		public static ICacheConcurrencyStrategy CreateCache(
+			string usage,
+			string name,
+			bool mutable,
+			Settings settings,
+			Func<string, ICache> regionCacheGetter)
+		{
+			if (usage == null || !settings.IsSecondLevelCacheEnabled) return null; //no cache
```

Comment:
It uses a cache "getter" parameter instead of directly the region cache because of this test, in order to avoid creating unneeded region caches.

Otherwise the "usage == null => no cache" rule would have to be moved into the session factory, instead of staying the responsibility of cache factory.